tsdb/store.go

```diff
@@ -957,6 +946,20 @@ func IsRetryable(err error) bool {
 	return true
 }
 
+// DecodeStorePath extracts the database and retention policy names
+// from a given shard or WAL path.
+func DecodeStorePath(shardOrWALPath string) (database, retentionPolicy string) {
```

Comment:
Maybe I'm overlooking something here, but why can't you just split on `/` and take the correct elements from the resulting `[]string`?

```go
parts := strings.Split(pth, "/")
// check parts is of appropriate length.
return parts[len(parts)-3], parts[len(parts)-2]
```